### DISCUSSION TITLE
Isn't this bad design

### DISCUSSION THREAD

**Discussion by Brentably** [vercel/next.js#50023] [Help]
### Summary

So I take issue with the next/jest package. It does not allow you to override the transformIgnorePatterns option from jest, even though normally the config overrides these. (see this line [https://github.com/vercel/next.js/blob/4b4925f4e8cf1d386ffa3f161c06fc60aa136d77/packages/next/src/build/jest/jest.ts#L180](https://github.com/vercel/next.js/blob/4b4925f4e8cf1d386ffa3f161c06fc60aa136d77/packages/next/src/build/jest/jest.ts#L180) )

I've been stuck on this bug where I need to override it, NOT APPEND (see [https://github.com/vercel/next.js/discussions/31152#discussioncomment-1796386](https://github.com/vercel/next.js/discussions/31152#discussioncomment-1796386))

Isn't this just straight up bad design?  Especially because the extent of the documentation is this folder: [https://github.com/vercel/next.js/tree/canary/examples/with-jest](https://github.com/vercel/next.js/tree/canary/examples/with-jest)

Which covers the cleanest, simplest, best-case-scenario for next/jest.

**Comment by icyJoseph** [comment]
Probably... I also have some issue with it, but there's a fix!

You can use the function form:

```js
const nextJest = require('next/jest')

const createJestConfig = nextJest({
  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
  dir: './',
})

// Add any custom config to be passed to Jest
const customJestConfig = {
  // Add more setup options before each test is run
  // setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  // if using TypeScript with a baseUrl set to the root directory then you need the below for alias' to work
  moduleDirectories: ['node_modules', '<rootDir>/'],
  testEnvironment: 'jest-environment-jsdom',
}

module.exports = async (...args) => {
  const fn = createJestConfig(customJestConfig)
  const res = await fn(...args)
  
  res.transformIgnorePatterns = res.transformIgnorePatterns.map(pattern => {
    if (pattern === '/node_modules/') {
      return '/node_modules(?!/excluded-module)/'
    }
    return pattern
  })
  
  return res
}
```

Taken from: https://github.com/vercel/next.js/discussions/34774#discussioncomment-2246460 where there's somewhat of an explanation for it.
